components/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountConnectorImpl.java

```diff
@@ -500,4 +517,48 @@ private void logWarnMessage(String userName1, String userName2, boolean user1Exi
         }
         log.warn(errorMsg);
     }
+
+    private boolean isOwnerHasAValidAssociation(String ownerUserName, String associatedUserName)
+            throws UserAccountAssociationException {
+
+        boolean isOwnerHasValidAssociation = false;
+        User associatedUser = getAssociatedUser(associatedUserName);
+        for (UserAccountAssociationDTO eachUserAccountAssociation : getAccountAssociationsOfUser(ownerUserName)) {
+            if (isSameAsTheAssociatedUser(associatedUser, eachUserAccountAssociation)) {
+                isOwnerHasValidAssociation = true;
+            }
+        }
+        return isOwnerHasValidAssociation;
+    }
+
+    private boolean isSameAsTheAssociatedUser(
+            User associatedUser, UserAccountAssociationDTO eachUserAccountAssociation) {
```

Comment:
eachUserAccountAssociation --> userAccountAssociation

Reply:
Fixed in 2bf8bed123a7a6907f221f3b0aded42b6cd5cb10